### ISSUE TITLE
Conflict with the Dynamic Prompts extension

### ISSUE THREAD

**Issue by OliverTempel** [artyfacialintelagent/CloneCleaner#1]
Settings:
Batch Count: 10
Positiv Prompt: RAW photo, best quality, high resolution), a perfect photo of a woman

Expected:
10 differtent Prompts:
Patricia from Uganda, (short-cut:1.2) bob-cut hair, RAW photo, best quality, high resolution), a ...
Riley from Hong Kong, medium (shoulder-length:1.2) grown-out pixie-cut hair, RAW photo, best quality, high resolution), a ...
...

Got:
Patricia from Uganda, (short-cut:1.2) bob-cut hair, RAW photo, best quality, high resolution), a ...
Patricia from Uganda, (short-cut:1.2) bob-cut hair, RAW photo, best quality, high resolution), a ...
...

**Comment by Gwynei** [comment]
Can confirm.

**Comment by artyfacialintelagent** [comment]
~~Unfortunately I can't reproduce this on my system~~. [EDIT: caused by an extension conflict, see comment below] I get different random tokens for each image whether I use batch count or batch size. It would be great if both of you could help me with more diagnostics:

* Is batch size working correctly, and the problem only appear with batch count?
* Please copy the whole version info line at the very bottom of the auto1111 page and paste it here.
* Also what OS are you on?
* And maybe also upload an image generated with CloneCleaner to [catbox.moe](https://catbox.moe/) so I can inspect the metadata.

Thanks and sorry about the problem!

**Comment by OliverTempel** [comment]
Version-Info:
`version: [v1.4.0](https://github.com/AUTOMATIC1111/stable-diffusion-webui/commit/394ffa7b0a7fff3ec484bcd084e673a8b301ccc8)  
python: 3.10.7
torch: 2.0.1+cu118  
xformers: N/A  
gradio: 3.32.0
checkpoint: [bfea7e18e2](https://google.com/search?q=bfea7e18e236471741f96ad5e987f214eb31ac75a923ca6ff0ef883d5d15f85e)`

OS: 
`windows 10 pro`

Extensions:
- CloneCleaner
- a1111-sd-webui-lycoris
- a1111-sd-webui-tagcomplete
- adetailer
- deforum-for-automatic1111-webui
- openpose-editor
- sd-dynamic-prompts
- sd-webui-3d-open-pose-editor
- sd-webui-additional-networks
- sd-webui-controlnet
- sd_civitai_extension
- stable-diffusion-webui-images-browser
- stable-diffusion-webui-rembg
- training-picker
- ultimate-upscale-for-automatic1111

[https://files.catbox.moe/7rz3o2.zip](url)
[https://files.catbox.moe/yqygiv.zip](url)

**Comment by OliverTempel** [comment]
i think i could help you with this error, disabled all extensions and enabled extension for extension and check if the error came back.

as soon as i enable "sd-dynamic-prompts" error came back. 

So please fix it as soon as possible. "sd-dynamic-prompts" is a very important extension :)

**Comment by songib** [comment]
yeah, I got the same issue atm. fix seed or random seed didn't solve the problem.
some people already mention "sd-dynamic-prompts", so probably using some exception to works similar like Adetailer with this exclusion "dynamic_prompting,dynamic_thresholding,wildcard_recursive,wildcards"

IDK much about programming stuff. so yeah. xd
Same seed
sd-dynamic-prompts is Enable
![grid-0001](https://github.com/artyfacialintelagent/CloneCleaner/assets/8655587/7331f905-eeb5-40f2-a78b-2692d1ac8af1)
sd-dynamic-prompts is Disable
![grid-0002](https://github.com/artyfacialintelagent/CloneCleaner/assets/8655587/40f478b6-11bc-4bea-b7c1-a4d11ee7dc98)

**Comment by GalaxyTimeMachine** [comment]
I have the same problem when using "Dynamic Prompts" extension.

**Comment by artyfacialintelagent** [comment]
Thanks everyone for reporting. I can reproduce the problem and can confirm that there is indeed a conflict with the Dynamic Prompts extension. Apparently that extension only reads the prompt from the first image of a batch. CloneCleaner works by modifying the individual prompts of a batch, and works correctly even when Dynamic Prompts is enabled. However, since DP runs after CloneCleaner (extensions run in alphabetical order I believe), DP clobbers the prompt changes that CloneCleaner has made.

Unfortunately that means that there is no way I can fix the conflict myself, but I will open an issue for Dynamic Prompts and ask them to play nice with other extensions (by reading all prompts instead of just the first of a batch).

So this means it is not yet possible to run CloneCleaner and Dynamic Prompts simultaneously, so you will have to enable/disable them depending on which you prefer to run for any given prompt. Once DP addresses the underlying issue the extensions should be compatible though, fingers crossed.
